### ISSUE TITLE
Next alpha release?

### ISSUE THREAD

**Issue by dbenninger** [arquillian/arquillian-extension-persistence#55]
Hi, 

I have been wondering if and when your next alpha release will be. There is one specific code change in the master branch that would solve a problem: 

Newer version of snakeyaml removed support for "Value Key Language-Independent Type" in yaml, see (https://github.com/Thinkofname/snakeyaml/issues/192). Your most recent version still somehow depends on that, meaning that it throws a NoSuchFieldError when working with a newer version of snakeyaml from the container. As I can see in your code, you already removed the reference to the VALUE field from your codebase on the master branch.

Exception details:
ava.lang.NoSuchFieldError: VALUE
at org.jboss.arquillian.persistence.dbunit.dataset.yaml.YamlDataSetProducer$1.addImplicitResolvers(YamlDataSetProducer.java:116)

Thanks

**Comment by bartoszmajsak** [comment]
Hi,

Thanks for pointing it out. I think we can push one next week.

**Comment by dbenninger** [comment]
That would be great!

Thanks for your quick reply..

**Comment by bartoszmajsak** [comment]
I would keep it open though until I take a closer look :)

**Comment by dbenninger** [comment]
@bartoszmajsak, any news on this one?

**Comment by bartoszmajsak** [comment]
I reserved time for releasing this small update this week. I guess it will be a new `Alpha`, as for going to `Final` version I would love to have more features incorporated. I will keep you posted.

**Comment by dbenninger** [comment]
that would be great. Also, could you do the alpha release of the core: https://github.com/arquillian/arquillian-extension-persistence

we need the fix for the bridge methods as well...

**Comment by mohlek** [comment]
When will be next week?

**Comment by bartoszmajsak** [comment]
:) Very soon! Apologies. Our goal is to release new version by Wed (including new 2.0.0-alpha version0

**Comment by mohlek** [comment]
Good to hear. I came across this project and was really amazed by `@UsingDataSet`. But apparently its not working as I need to assign a Datasource via `@DataSource` because i have multiple persistence units. I hope the next release will fix my problems :)

**Comment by bartoszmajsak** [comment]
If you can open an issue describing your challenges we will see how we can improve :)

**Comment by mohlek** [comment]
Basically its #48 :)

**Comment by bartoszmajsak** [comment]
Busy weekend that is ;)

**Comment by madoar** [comment]
Are there any news about a new release of this extension?
The last release on maven (Alpha 7) is from May 2014, which is now almost 3 years ago.
It would be really nice if we could get a version update.

**Comment by bartoszmajsak** [comment]
If you look closer in the repository you'd notice there is a 2.0.0 branch. We are working very hard on bringing this new, revamped version as well as releasing the final version of 1.0.0 supporting the pre Java EE 7 environments. Stay tuned. It's coming very very soon.

**Comment by madoar** [comment]
Thanks for the fast answer. Yes I saw the branch ~10 minutes after posting.
Can you give us a short summary about the new features/improvements of the 2.0.0 version, compared to the 1.0.0-Alpha7 one?

**Comment by bartoszmajsak** [comment]
Sure, once I come back from Easter break we will make a blog post on arquillian.org. Mobile only till Monday evening :)

**Comment by madoar** [comment]
Okay, thanks for the feedback and enjoy your Easter break. ;)

**Comment by bartoszmajsak** [comment]
We won't do a blog post on [http://arquillian.org](arquillian.org) yet, but we would like to share with all of you our plan for next month to bring you both `1.0.0.Final` and brand new `2.0.0` version (we have already released two alphas for the latter).

We would like to keep "Arquillian Persistence Extension" focusing on JPA and DBUnit in 1.x.y version, working on older JVMs (and thus supporting older Java EE containers). We will ship several important features towards this release:

  * Scriptable data sets
  * Ability to use multiple data sources in `@Using/@ShouldMatchDataset`
  * Major bug fixes reported in JIRA
  * Standalone support (non in-container tests)

All above will also be available in new version. With `2.0.0` we rebrand our tool and simply call it APE, as we've been internally calling it for years. The reason is that what we are working on goes beyond "persistence" which is primarily associated with JPA in the Java world. We are going to provide unified developer experience for seeding other data storages (including NoSQL and more - we count on your feedback here!). Also, dealing with test data is tedious and boring, so let's have an APE doing it for us! Our 30000 feet master plan boils down to:

  * Unified programmatic (dsl) and declarative (annotations) way of seeding data
  * Support for SQL, JPA, NoSQL and more!
  * Leveraging more of containers than ever using our awesome [Arquillian Cube](https://github.com/arquillian/arquillian-cube)

Having said all of that - please give us a bit more time to make it happen, as it's not a trivial effort. If you have any suggestions we are thrilled to hear them. This will help us shape APE in the way YOU want it!

**Comment by TheWriterAddict** [comment]
Excellent ! 

Support @UsingDataSet before deploy would be nice !

**Comment by mohlek** [comment]
@bartoszmajsak any chance to get the 1.0.0.Final?
How about pushing the 2.0 Alpha releases to maven?

**Comment by bartoszmajsak** [comment]
Alpha releases of 2.0.0 are in maven already, under the new name and structure http://search.maven.org/#search|ga|1|arquillian-ape

As for 1.0.0.Final I'm working now on https://issues.jboss.org/browse/ARQ-1645

Once this is done I will release.

**Comment by mohlek** [comment]
Ah sry I didn't see that.
I will take a look at 2.0.0 :)

**Comment by bartoszmajsak** [comment]
For anyone interested I briefly described the migration process:

https://github.com/arquillian/arquillian-extension-persistence/blob/2.0.0/README.adoc#2-migration-from-arquillian-persistence-extension-1x-to-2x

**Comment by gsomeshrao** [comment]
hi
@bartoszmajsak Have some queries for you 

1. Any update on https://issues.jboss.org/browse/ARQ-1645 , regarding the support for multiple datasources ?

2. I have gone through the migration doc , but it has very less info.
https://github.com/arquillian/arquillian-extension-persistence/blob/2.0.0/README.adoc#2-migration-from-arquillian-persistence-extension-1x-to-2x
Does the fix for ARQ-1645 work only with Arquillian ape 2.x version only ?

3. I see 2.0.0.alpha.x is the latest in maven. Any arquillian-bom available for the same to support ARQ-1645

4. As described in the migration doc, I was using the dependency "arquillian-ape-sql-container-dbunit" of 2.0.0-alpha.4 version. Given below are the dependencies i hav in our pom.xml for testing the support for multiple datasources. Pls let me know if , this the correct combination of dependencies?
----------------
		<dependency>
			<groupId>org.arquillian</groupId>
			<artifactId>arquillian-universe</artifactId>
			<version>1.1.13.5</version>
			<type>pom</type>
		</dependency>
		<dependency>
			<groupId>org.jboss.arquillian.junit</groupId>
			<artifactId>arquillian-junit-container</artifactId>
			<version>1.1.13.Final</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.arquillian.ape</groupId>
			<artifactId>arquillian-ape-sql-container-dbunit</artifactId>
			<version>2.0.0-alpha.4</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.eu.ingwar.tools</groupId>
			<artifactId>arquillian-suite-extension</artifactId>
			<version>1.1.4</version>
		</dependency>
-----------------
I am gettting ClassCastException as given below, after changing to "arquillian-ape-sql-container-dbunit" from "arquillian-persistence-dbunit". 

java.lang.ClassCastException: org.arquillian.ape.rdbms.core.event.ApplyScriptsBeforeTest cannot be cast to org.arquillian.ape.rdbms.dbunit.event.CompareDBUnitData
	at org.arquillian.ape.rdbms.dbunit.DBUnitDataHandler.compare(DBUnitDataHandler.java:58)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:99)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:81)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:145)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:116)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:67)
	at org.arquillian.ape.rdbms.core.lifecycle.CustomScriptsExecutor.executeScriptsBeforeTest(CustomScriptsExecutor.java:65)
	at org.arquillian.ape.rdbms.core.lifecycle.CustomScriptsExecutor.executeBeforeTest(CustomScriptsExecutor.java:52)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:99)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:81)
	at org.arquillian.ape.rdbms.dbunit.DBUnitDataStateLogger.beforePersistenceTest(DBUnitDataStateLogger.java:94)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.arquillian.ape.rdbms.core.lifecycle.SchemaCreationScriptsExecutor.createSchema(SchemaCreationScriptsExecutor.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:130)
	at sun.reflect.GeneratedMethodAccessor102.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:73)
	at sun.reflect.GeneratedMethodAccessor96.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:92)
	at sun.reflect.GeneratedMethodAccessor98.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.arquillian.ape.rdbms.dbunit.DBUnitPersistenceTestLifecycleHandler.createDatasets(DBUnitPersistenceTestLifecycleHandler.java:111)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.arquillian.ape.rdbms.core.lifecycle.ErrorCollectorHandler.createErrorCollector(ErrorCollectorHandler.java:37)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:88)
	at org.arquillian.ape.rdbms.dbunit.DBUnitPersistenceTestLifecycleHandler.provideDatabaseConnectionAroundBeforePersistenceTest(DBUnitPersistenceTestLifecycleHandler.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
----------------

Appreciate your feedback on the above...

**Comment by bartoszmajsak** [comment]
Hi @gsomeshrao. Could you help me with following?

  * reformat your comment as it's a bit hard to follow the stacktrace
  * provide all relevant info about your environment
  * if possible - a small reproducer project?

Thanks!

**Comment by gsomeshrao** [comment]
@bartoszmajsak We are working on a project that has multiple datasources.
Can you provide me a test example of configuring and running with multiple datasources?
